File: Android/app/src/main/java/com/smartsolar/microgrid/model/UpdateProsumerRequest.kt
package com.smartsolar.microgrid.model

/**
 * Request body used to update a Prosumer's profile.
 */
data class UpdateProsumerRequest(

    // Updated name.
    val name: String,

    // Updated email.
    val email: String,

    // Updated phone number.
    val phone: String,

    // Updated address.
    val address: String
)

File: Backend/SmartSolarMicrogrid.API/DTOs/Prosumers/UpdateProsumerRequest.cs
using System.ComponentModel.DataAnnotations;

namespace SmartSolarMicrogrid.API.DTOs.Prosumers;

/// information that a Prosumer can update in their profile.
public class UpdateProsumerRequest
{
    [Required]
    public string Name { get; set; } = string.Empty;

    [Required]
    [EmailAddress]
    public string Email { get; set; } = string.Empty;

    [Required]
    public string Phone { get; set; } = string.Empty;

    [Required]
    public string Address { get; set; } = string.Empty;
}